### PR TITLE
Prevent custom_header with non-ASCII characters (#1840)

### LINKED ISSUE #1840: Prevent custom_header with accents
Expected Behavior
-----------------
People shouldn’t be able to set a custom_header with an accent, which makes Sympa crash.

Current Behavior
----------------
People can set a custom_header with an accent, which makes Sympa crash.

Possible Solution
-----------------
We could set a [`pattern`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text#pattern) attribute to the input field, like `^[a-zA-Z0-9]*$`, or set a similar validation in the backend.

Context
-------
A user has set `X-Expéditeur` as custom_header, which made sympa_msg crash, preventing to keep processing other mails.

Crash log:
```
2024-05-14T15:59:11.054874+02:00 rod3 sympa_msg[2057866]: err main::#242 > Sympa::Spindle::spin#95 > Sympa::Spindle::TransformOutgoing::_twist#105 > Sympa::Message::add_header#399 > Mail::Header::add#472 > Mail::Header::_fmt_line#163 > Carp::croak#289 DIED: Bad RFC822 field name 'X-Expéditeur'  at /home/sympa/bin/Sympa/Message.pm line 399.
```

### PR DESCRIPTION
**Pull request by ikedas**
The `custom_header` should not contain non-ASCII characters.
Additionally, some minor bugs are fixed.

This may fix #1840.

### CODE REVIEW

File: src/lib/Sympa/Regexps.pm [hunk 1]
```diff
@@ -47,6 +47,8 @@ use constant email =>
 use constant family_name => qr'[a-z0-9][a-z0-9\-\.\+_]*';
 ## Allow \s for template names
 use constant template_name => qr'[a-zA-Z0-9][a-zA-Z0-9\-\.\+_\s]*';
+# cf. RFC5322, 2.2.
+use constant header_field_name => qr'[!-9;-~]+';
```
Comment (ldidry) [inline src/lib/Sympa/Regexps.pm:51]: I’m not familiar with the interval `!-9`, but I guess it contains a lot of ascii chars and no accentuated letters.

I trust your regex skills :slightly_smiling_face:

File: src/lib/Sympa/Config/Schema.pm [hunk 2]
```diff
@@ -1353,7 +1353,7 @@ our %pinfo = (
             'Header fields to be removed before message distribution',
         gettext_comment =>
             "The removal happens after Sympa's own header fields are added; therefore, it is a convenient way to remove Sympa's own header fields (like \"X-Loop:\" or \"X-no-archive:\") if you wish.",
-        format     => '\S+',
+        format_s   => '$header_field_name(:.+)?',
```
Comment (ldidry) [inline src/lib/Sympa/Config/Schema.pm:1356]: The previous regex prevented space, but not this one. Is this normal?

Comment (ikedas) [reply src/lib/Sympa/Config/Schema.pm:1356]: The code in [TransformOutgoing.pm](https://github.com/sympa-community/sympa/blob/55598aebb2c0ba4b8b5be733563cad89973cf2fc/src/lib/Sympa/Spindle/TransformOutgoing.pm#L145-L147) looks assuming a parameter value such as `Header-Field-Name: value`.  I think the white space(s) between colon and value is optional anyway.

File: src/lib/Sympa/Regexps.pm (hunk 1 above)
Comment (ikedas) [reply src/lib/Sympa/Regexps.pm:51]: `[!-9;-~]` represents "printable ASCII character except colon".  I only hope I have not betrayed your trust 😓 .

File: src/lib/Sympa/Config/Schema.pm (hunk 2 above)
Comment (ldidry) [reply src/lib/Sympa/Config/Schema.pm:1356]: :+1:

File: src/lib/Sympa/Regexps.pm (hunk 1 above)
Comment (ldidry) [reply src/lib/Sympa/Regexps.pm:51]: You didn’t! :ok_hand:

**Review by ldidry — APPROVED** [review]

**Pull request merged**